examples/dreambooth/train_dreambooth_lora_sdxl.py

```diff
@@ -1083,6 +1100,17 @@ def load_model_hook(models, input_dir):
             args.learning_rate * args.gradient_accumulation_steps * args.train_batch_size * accelerator.num_processes
         )
 
+    # Make sure the trainable params are in float32.
+    if args.mixed_precision == "fp16":
+        models = [unet]
+        if args.train_text_encoder:
+            models.extend([text_encoder_one, text_encoder_two])
+        for model in models:
+            for param in model.parameters():
+                # only upcast trainable parameters (LoRA) into fp32
+                if param.requires_grad:
+                    param.data = param.to(torch.float32)
```

Comment:
We do it just before assigning the parameters to the optimizer to avoid any consequences.

Reply:
In a follow-up PR, I can wrap this utility into a function and move to `training_utils.py`. It's shared by a number of scripts.